### DISCUSSION TITLE
List of functions

### DISCUSSION THREAD

**Discussion by larmarange** [quarto-dev/quarto-cli#2747] [Q&A]
When writing a manual/book, the autolink feature is very great. 

But it would be very useful to be able to easily create an index of functions indicating where they are mentioned in the book.

Something similar to https://larmarange.github.io/analyse-R/index-des-fonctions.html  It helps the reader to identify in which chapter that function is presented/used.

**Comment by mcanouil** [comment]
How did you do it in the website you are referencing?
Maybe {downlit} has a feature allowing to do that (list of back reference). 🤔

**Reply by larmarange** [reply]
It was a custom code (cf. https://github.com/larmarange/analyse-R/blob/gh-pages/index-des-fonctions.Rmd ). Also, that website was using custom markup for functions.

It doesn't seem that `downlit` has such a function.

**Reply by mcanouil** [reply]
Did you try to reuse your code to achieve the same results, did it work?

In some aspects, this feature request is related to the glossary one (see #1697)

**Reply by larmarange** [reply]
Yes, it is related in some aspects to the question of the glossary. The idea would be to have something automatic using the syntax recognized by `downlit`.

> Did you try to reuse your code to achieve the same results, did it work?

No and it could be working as it. My code was using some custom markup.

I have the feeling that such functionality would probably require an evolution of `downlit` where `downlit` would have an option to save a tibble with the list of topics, their type (function, package, vignette...), the computed links and their position (at least which file).

**Reply by larmarange** [reply]
https://github.com/r-lib/downlit/issues/161

**Reply by mcanouil** [reply]
I also think it would be better to have such feature implemented in `downlit`.
